src/snowflake/connector/connection.py

```diff
@@ -1107,8 +1107,10 @@ def cmd_query(
             # binding parameters. This is for qmarks paramstyle.
             data["bindings"] = binding_params
         if not _no_results:
-            # not an async query
-            data["queryContext"] = self.get_query_context()
+            # not an async query. Here queryContextDTO should be a dict object field, same with `parameters` field
+            queryContext = self.get_query_context()
+            if queryContext is not None:
+                data["queryContextDTO"] = queryContext
```

Comment:
this change means that some responses to GS will not contain `queryContextDTO` field. Is that okay?

Reply:
I think they are equivalent when GS convert JSON back to the query context object. But I restored to the original version to avoid unnecessary changes. I have tested using my e2e queries and their query context entries look good.